Code/Engine/Core/Scripting/ScriptComponent.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include <Core/Scripting/ScriptClassResource.h>
+#include <Core/World/EventMessageHandlerComponent.h>
+#include <Foundation/Types/RangeView.h>
+
+using ezScriptComponentManager = ezComponentManager<class ezScriptComponent, ezBlockStorageType::FreeList>;
+
+class EZ_CORE_DLL ezScriptComponent : public ezEventMessageHandlerComponent
+{
+  EZ_DECLARE_COMPONENT_TYPE(ezScriptComponent, ezEventMessageHandlerComponent, ezScriptComponentManager);
+
+  //////////////////////////////////////////////////////////////////////////
+  // ezComponent
+
+protected:
+  virtual void SerializeComponent(ezWorldWriter& stream) const override;
+  virtual void DeserializeComponent(ezWorldReader& stream) override;
+  virtual void Initialize() override;
+  virtual void Deinitialize() override;
+  virtual void OnActivated() override;
+  virtual void OnDeactivated() override;
+  virtual void OnSimulationStarted() override;
+
+  /*virtual bool OnUnhandledMessage(ezMessage& msg, bool bWasPostedMsg) override;
+  virtual bool OnUnhandledMessage(ezMessage& msg, bool bWasPostedMsg) const override;
+
+  bool HandleUnhandledMessage(ezMessage& msg, bool bWasPostedMsg);*/
```

Comment:
remove

Reply:
Message handling is not implemented yet so I will leave them in there until I know whether I need them or not.